### PR TITLE
Add 'omitemptycheckstruct' tag

### PR DESCRIPTION
**Pull request by akalin-keybase**
This tag, when used with the 'omitempty' tag, makes the
empty check work for structs.

### CODE REVIEW

File: codec/codec_test.go [hunk 1]
```diff
@@ -1411,6 +1411,46 @@ func doTestEncUnknownFields(t *testing.T, h Handle) {
 	}
 }
 
+func doTestOmitEmpty(t *testing.T, h Handle) {
+	type O1 struct {
```
Comment (strib) [inline codec/codec_test.go]: I'd prefer not starting var names with a capital `O`.  Obviously it's not ambiguous since var names can't start with numbers, but I still think using `S` or something would be easier to read.

File: codec/codec_test.go [hunk 2]
```diff
@@ -1411,6 +1411,46 @@ func doTestEncUnknownFields(t *testing.T, h Handle) {
...
+		t.Fatal(err)
+	}
+
+	expectedM := map[string]interface{}{
+		"A": map[interface{}]interface{}{
+			"M": nil,
+		},
+	}
+
+	// Decoded map should have A but not B.
```
Comment (strib) [inline codec/codec_test.go:1448]: Maybe also add a test where `B` isn't empty?

**Comment by strib** [comment]
Looks good to me, just minor test suggestions.

File: codec/codec_test.go (hunk 1 above)
Comment (akalin-keybase) [inline codec/codec_test.go]: Done.

File: codec/codec_test.go (hunk 2 above)
Comment (akalin-keybase) [inline codec/codec_test.go:1448]: Done.
